### PR TITLE
add glob path parser

### PR DESCRIPTION
**Pull request by liudonghua123**
I changed the path argument parser using glob for cross-platform and regexp matching. The changes are backward compatible.
And now you can use the following pattern in any OS and shell intepreters.
```shell
# use glob style, recommend!
flutter pub run intl_translation:extract_to_arb --output-dir=lib/l10n lib/**.dart
# use old style, maybe tredious and error-prone
flutter pub run intl_translation:extract_to_arb --output-dir=lib/l10n lib/localization.dart lib/app.dart

# use glob style, recommend!
flutter pub run intl_translation:generate_from_arb --output-dir=lib/l10n --no-use-deferred-loading lib/**.dart lib/l10n/intl_*.arb
# use old style, maybe tredious and error-prone
flutter pub run intl_translation:generate_from_arb --output-dir=lib/l10n --no-use-deferred-loading lib/localization.dart lib/app.dart lib/l10n/intl_messages.arb lib/l10n/intl_en.arb lib/l10n/intl_zh.arb

```

### CODE REVIEW

**Comment by liudonghua123** [comment]
@googlebot I signed it!

**Comment by lzhuor** [comment]
@liudonghua123 Do you mind rebasing this branch? Thank you!

**Comment by liudonghua123** [comment]
@lzhuor Hi, I fixed the conflicts.

File: bin/generate_from_arb.dart [hunk 1]
```diff
@@ -87,10 +88,27 @@ main(List<String> args) {
...
+      .where((x) => x.endsWith(".dart"))
+      .toList();
+  var jsonFiles = args
+      .map((x) => Glob(x).listSync().map((file) => file.path))
+      .expand((i) => i)
+      .where((x) => x.endsWith(".arb"))
+      .toList();
   dartFiles.addAll(linesFromFile(sourcesListFile));
   jsonFiles.addAll(linesFromFile(translationsListFile));
+  // remove duplicates
```
Comment (mosuem) [inline bin/generate_from_arb.dart:105]: Why could there be duplicates?

**Comment by mosuem** [comment]
Using wildcards works already now.

**Pull request closed without merging**